.github/workflows/unit_tests.yml

```diff
@@ -10,45 +10,25 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Update the system
         run: |
           sudo apt-get update -y
 
       - name: Install dependencies
         run: |
-          sudo apt-get install -y shunit2 kcov bash git \
-          python-docutils dash graphviz python3-sphinx
+          sudo apt-get install -y shunit2 git
 
       - name: Prepare for tests
         run: |
           ./run_tests.sh prepare
-          mkdir kcov_out/
 
-      - name: Unit tests
+      - name: Setup mock git user
         run: |
           git config --global user.email "kw@kworkflow.net"
           git config --global user.name "Kworkflow"
-          git init
-          git commit --allow-empty -m "Test commit 1"
-          git commit --allow-empty -m "Test commit 2"
-          ./run_tests.sh
```

Comment:
I think deleting these lines caused some issues with the tests.
@kwy95 any idea here?

Reply:
I managed to solve this by add more depth to the checkout action :)